microsoft/PowerToys
# [Build] Centralize .NET NuGet Package Versions

## Summary of the Pull Request
Enables Centralized Package Management for .NET projects. All NuGet package versions are managed within `Directory.Packages.props`, removing the need for individual versions to be defined within a project. It fully incorporates with Visual Studio and will automatically update the version in a single place / add a new entry when a new package is added to a project.

Currently only .NET packages are supported. 


## PR Checklist



## Detailed Description of the Pull Request / Additional comments


## Validation Steps Performed

- Run `ls -Recurse -include bin,obj,ARM64,x86,x64,packages,Release,Debug | rmdir -recurse` in PowerShell from the root PowerToys repo directory to remove all packages and existing compiled binaries
- Execute a NuGet Package Restore
- Rebuild the entire solution
- Temporarily update a package version and verify `Directory.Packages.props` is updated with a version matching
- Temporarily add a package to a project and verify `Directory.Packages.props` is updated with a new entry referencing the package and version.
